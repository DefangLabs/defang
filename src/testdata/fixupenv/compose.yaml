services:
  Mistral:
    image: "mistral:latest"
    ports:
      - mode: host
        target: 8000
  ui:
    image: "ui:latest"
    environment:
      - "API_URL=http://Mistral:8000"
      - "SENSITIVE_DATA"
  ingress-service:
    image: "somedb:latest"
    ports:
      - mode: ingress
        target: 5432
  use-ingress-service:
    image: "service:latest"
    environment:
      - "UI_URL=ingress-service:5432"
  fixup-args:
    build:
      args:
        - "API_URL=http://Mistral:8000"
  refer-self:
    image: "service:latest"
    environment:
      - "CONFIG3=refer-self"
    ports:
      - mode: ingress
        target: 5678
