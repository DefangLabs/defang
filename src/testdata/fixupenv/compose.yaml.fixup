{
  "Mistral": {
    "command": null,
    "entrypoint": null,
    "image": "mistral:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 8000,
        "protocol": "tcp"
      }
    ]
  },
  "fixup-args": {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "API_URL": "http://mock-mistral:8000"
      }
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  },
  "ingress-service": {
    "command": null,
    "entrypoint": null,
    "image": "somedb:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5432,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "refer-self": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "CONFIG3": "refer-self.fixupenv.tenant2.defang.app"
    },
    "image": "service:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5678,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  "ui": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "API_URL": "http://mock-mistral:8000",
      "SENSITIVE_DATA": null
    },
    "image": "ui:latest",
    "networks": {
      "default": null
    }
  },
  "use-ingress-service": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "UI_URL": "ingress-service.fixupenv.tenant2.defang.app:5432"
    },
    "image": "service:latest",
    "networks": {
      "default": null
    }
  }
}