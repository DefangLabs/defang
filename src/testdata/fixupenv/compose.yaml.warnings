 ! service "Mistral": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "env-in-config": environment variable(s) ["CONFIG1"] will use the `defang config` value instead of adjusted service name
 ! service "env-in-config": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "fixup-args": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "fixup-args": service name was adjusted: build argument "API_URL" assigned value "http://mock-mistral:8000"
 ! service "ingress-service": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "ingress-service": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "ui": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "ui": service name was adjusted: environment variable "API_URL" assigned value "http://mock-mistral:8000"
 ! service "use-ingress-service": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "use-ingress-service": service name in the environment variable "DB_URL" was not adjusted; only references to other services with port mode set to 'host' will be fixed-up
 * Packaging the project files for fixup-args at .
