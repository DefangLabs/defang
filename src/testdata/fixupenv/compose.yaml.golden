name: fixupenv
services:
  Mistral:
    image: mistral:latest
    networks:
      default: null
    ports:
      - mode: host
        target: 8000
        protocol: tcp
  fixup-args:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        API_URL: http://Mistral:8000
    networks:
      default: null
  ingress-service:
    image: somedb:latest
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5432
        protocol: tcp
  refer-self-build-arg:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        CONFIG4: refer-self-env
    networks:
      default: null
  refer-self-env:
    environment:
      CONFIG3: refer-self-env
    image: service:latest
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5678
        protocol: tcp
  ui:
    environment:
      API_URL: http://Mistral:8000
      SENSITIVE_DATA: null
    image: ui:latest
    networks:
      default: null
  use-ingress-service:
    environment:
      UI_URL: ingress-service:5432
    image: service:latest
    networks:
      default: null
networks:
  default:
    name: fixupenv_default
